llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -794,12 +794,17 @@ static void legalizeAndOptimizeInductions(VPlan &Plan) {
         WideIV->getDebugLoc(), Builder);
 
     // Update scalar users of IV to use Step instead.
-    if (!HasOnlyVectorVFs)
+    if (!HasOnlyVectorVFs) {
```

Comment:
Is it worth adding an assert here that the plan doesn't have a scalable VF? i.e.

```
  if (!HasOnlyVectorVFs) {
    assert(!Plan.hasScalableVF() && ...);
...
```